tachyon/math/base/semigroups.h

```diff
@@ -215,6 +215,22 @@ class MultiplicativeSemigroup {
     return ret;
   }
 
+  constexpr static std::pmr::vector<MulResult> GetSuccessivePowersPmr(
+      size_t size, const G& generator, const G& c = G::One()) {
+    std::pmr::vector<MulResult> ret(size);
+    base::Parallelize(
+        ret, [&generator, &c](absl::Span<G> chunk, size_t chunk_offset,
+                              size_t chunk_size) {
+          MulResult pow = generator.Pow(chunk_offset * chunk_size);
+          if (!c.IsOne()) pow *= c;
+          for (G& v : chunk) {
+            v = pow;
+            pow *= generator;
```

Comment:
Please add `CHECK(!chunk.empty())` since if the `chunk.size() == 0` it will run through very long loop.

Please apply this trick to `GetSuccessiveScalarMuls()`, `EvaluatePartialLagrangeCoefficients()`, `DivideByVanishingPolyInPlace()` and `ComputeHQuery()`, too!

Reply:
I don't think it's possible to have empty chunk in `base::Parallelize`